linker-interface/shared/src/main/scala/org/scalajs/linker/interface/Report.scala

```diff
@@ -0,0 +1,155 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.linker.interface
+
+import java.io._
+
+import org.scalajs.ir.ScalaJSVersions
+
+import org.scalajs.linker.interface.unstable.ReportImpl
+
+/** Information about a linker run. */
+abstract class Report private[interface] {
```

Comment:
When we have a visibility modifier for the constructor, we always write the `()`:
```suggestion
abstract class Report private[interface] () {
```